src/helpers/buffers/ordering_sender.rs

```diff
@@ -197,6 +214,23 @@ impl Waiting {
     fn wake(&self, i: usize) {
         self.shard(i).wake(i);
     }
+
+    fn get_waiting_messages(&self) -> Vec<usize> {
+        let mut waiting_messages = self.shards[0].lock().unwrap().get_waiting_messages();
```

Comment:
I am not sure I understand how it works now. This returns messages waiting to be sent, so from the protocol's standpoint everything is done correctly - they sent the message. What we are interested at is messages that infra hasn't received from protocols yet (gaps)

Reply:
We want to figure out what are missing - I get that by subtracting the "not missing messages" from the set of all messages.